lib/benchpark/runtime.py

```diff
@@ -0,0 +1,147 @@
+from contextlib import contextmanager
+import os
+import pathlib
+import shlex
+import subprocess
+import sys
+import yaml
+
+DEBUG = True
+
+
+def debug_print(message):
+    if DEBUG:
+        print("(debug) " + str(message))
+
+
+@contextmanager
+def working_dir(location):
+    initial_dir = os.getcwd()
+    try:
+        os.chdir(location)
+        yield
+    finally:
+        os.chdir(initial_dir)
+
+
+def git_clone_commit(url, commit, destination):
+    run_command(f"git clone -c feature.manyFiles=true {url} {destination}")
```

Comment:
Does the `--depth=2` setting make it easy to check out other branches and pull full history if needed? If not, I'd prefer to keep the full depth.

Reply:
Yes, that's specifically why I suggested `depth=2` instead of `depth=1`. With `depth=1` git sets things that make it impossible to checkout other branches without modifying config. With `depth=2` we just need to fetch (to get the missing refs) before we can checkout other branches.

Reply:
Do you mind if this happens in a later PR?

Reply:
Yep fine with that